Does not compile with kernel 5.6

```
+ cd /scratch/build-kmod-pciedummy
+ sudo -H -E -u msk_jenkins make install DESTDIR=../install
In file included from /scratch/source/mtcadummy.h:39,
                 from /scratch/source/mtcadummy_drv.c:1:
/scratch/source/pciedev_io.h:60:18: error: field 'start_time' has incomplete type
   60 |   struct timeval start_time;
      |                  ^~~~~~~~~~
/scratch/source/pciedev_io.h:61:18: error: field 'stop_time' has incomplete type
   61 |   struct timeval stop_time;
      |                  ^~~~~~~~~
make[6]: *** [/usr/src/linux-5.6.0-1/scripts/Makefile.build:269: /scratch/source/mtcadummy_drv.o] Error 1
make[5]: *** [/usr/src/linux-5.6.0-1/Makefile:1698: /scratch/source] Error 2
make[4]: *** [../../../linux-5.6.0-1/Makefile:180: sub-make] Error 2
make[3]: *** [Makefile:9: all] Error 2
make[2]: *** [CMakeFiles/kmod.dir/build.make:74: CMakeFiles/kmod] Error 2
make[1]: *** [CMakeFiles/Makefile2:93: CMakeFiles/kmod.dir/all] Error 2
make: *** [Makefile:158: all] Error 2
```

--- a/mtcadummy_proc.c
+++ b/mtcadummy_proc.c
@@ -91,6 +91,7 @@ static int mtcadummy_proc_open(struct inode* inode, struct file* file) {
   return seq_open(file, &mtcadummy_seq_operations);
 }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
 /*
  * Create a set of file operations for the proc file.
  */
@@ -99,6 +100,14 @@ static struct file_operations mtcadummy_proc_opperations = {.owner = THIS_MODULE
     .read = seq_read,
     .llseek = seq_lseek,
     .release = seq_release};
+#else
+static struct proc_ops mtcadummy_proc_opperations = {
+    .proc_open = mtcadummy_proc_open,
+    .proc_read = seq_read,
+    .proc_lseek = seq_lseek,
+    .proc_release = seq_release
+};
+#endif
 
 /*
  * Actually create (and remove) the /proc file(s). They cannot be static because

--- a/mtcadummy_rw_no_struct.c
+++ b/mtcadummy_rw_no_struct.c
@@ -76,6 +76,7 @@ int checkAndCalculateTransferInformation(
     case 5:
       transferInformation->barStart = 0;
       transferInformation->barSizeInBytes = 0;
+      break;
 
     default:
       printk("MTCADUMMY_WRITE_NO_STRUCT: Invalid bar number %d\n", transferInformation->bar);

--- a/pciedev_io.h
+++ b/pciedev_io.h
@@ -3,6 +3,7 @@
 
 #include <linux/ioctl.h> /* needed for the _IOW etc stuff used later */
 #include <linux/types.h>
+#include <linux/version.h>
 
 #define RW_D8 0x0
 #define RW_D16 0x1
@@ -56,9 +57,24 @@ struct device_ioctrl_dma {
 };
 typedef struct device_ioctrl_dma device_ioctrl_dma;
 
-struct device_ioctrl_time {
-  struct timeval start_time;
-  struct timeval stop_time;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
+
+typedef struct timeval pcieuni_timeval;
+
+#else
+
+struct pcieuni_timeval {
+	long long tv_sec;
+	long long tv_usec;
+};
+
+typedef struct pcieuni_timeval pcieuni_timeval;
+#endif
+
+struct device_ioctrl_time  {
+
+        pcieuni_timeval   start_time;
+        pcieuni_timeval   stop_time;
 };
 typedef struct device_ioctrl_time device_ioctrl_time;
 

